src/app/core/tools/PanTool.ts

```diff
@@ -8,17 +8,21 @@ import {CircuitInfo} from "core/utils/CircuitInfo";
 
 import {Tool}        from "core/tools/Tool";
 
+import {Component}   from "core/models";
+
 export const PanTool: Tool = (() => {
     let isDragging = false;
     return {
-        shouldActivate(event: Event, {input}: CircuitInfo): boolean {
+        shouldActivate(event: Event, {input, selections}: CircuitInfo): boolean {
             // Activate if the user just pressed the "option key"
             //  or if the user began dragging with either 2 fingers
             //                           or the middle mouse button
-            //  or if the user pressed one of of the arrow keys
-            return (event.type === "keydown"   && ((event.key === OPTION_KEY) ||  
+            //  or if the user pressed one of of the arrow keys while no components are selected
+            return ((event.type === "keydown" && 
+                  !((selections.all((c) => c instanceof Component) && selections.get().length > 0)) && 
```

Comment:
Wait why do we need all of the selections to be components for the **Pan** tool?

Reply:
To my understanding, this checks to make sure that arrow keys don't pan when components are selected (since it should be translating those components instead)

Reply:
This should be done implicitly by the TranslateTool being before the PanTool in the list of tools (https://docs.google.com/drawings/d/11e6BiEQTNM3IdJawx4hk_HOe3uSZYisY_ICu5q6E8cY/edit?usp=sharing)

Reply:
That doesn't quite seem to do it. Removing the `!((selections.all((c) => c instanceof Component) && selections.get().length > 0))` condition causes the arrow keys to not translate at all, only pan. Removing just the `(selections.all((c) => c instanceof Component)` part of the condition allows for translating, but if just ports are selected, for example, then panning is blocked even though no translating is happening.

Reply:
You will have to put TranslateTool before the PanTool in the list of tools for it to work lol